kms-message/test/test_kms_kmip_request.c

```diff
@@ -0,0 +1,216 @@
+   uint32_t actual_len;
+   uint8_t expected_bytes[] = {REGISTER_SECRETDATA_REQUEST};
+   uint32_t expected_len = sizeof (expected_bytes);
+
+   req = kms_kmip_request_register_secretdata_new (
+      NULL, secret_data, sizeof (secret_data));
+   ASSERT_REQUEST_OK (req);
+
+   actual_bytes = kms_request_to_bytes (req, &actual_len);
+   ASSERT (actual_bytes != NULL);
+
+   ASSERT_CMPBYTES (expected_bytes, expected_len, actual_bytes, actual_len);
+
+   kms_request_destroy (req);
+}
+
+void
+kms_kmip_request_register_secretdata_invalid_test (void)
+{
+#define SECRET_DATA_INVALID_LENGTH 95
+
+   kms_request_t *req;
+   uint8_t secret_data[SECRET_DATA_INVALID_LENGTH] = {0};
+
+   req = kms_kmip_request_register_secretdata_new (
+      NULL, secret_data, SECRET_DATA_INVALID_LENGTH);
+   ASSERT_REQUEST_ERROR (req, "SecretData length");
+
+   kms_request_destroy (req);
+}
+
+
+/*
+<RequestMessage tag="0x420078" type="Structure">
+ <RequestHeader tag="0x420077" type="Structure">
+  <ProtocolVersion tag="0x420069" type="Structure">
+   <ProtocolVersionMajor tag="0x42006a" type="Integer" value="1"/>
+   <ProtocolVersionMinor tag="0x42006b" type="Integer" value="4"/>
+  </ProtocolVersion>
+  <BatchCount tag="0x42000d" type="Integer" value="1"/>
+ </RequestHeader>
+ <BatchItem tag="0x42000f" type="Structure">
+  <Operation tag="0x42005c" type="Enumeration" value="10"/>
+  <RequestPayload tag="0x420079" type="Structure">
+   <UniqueIdentifier tag="0x420094" type="TextString"
+value="7FJYvnV6XkaUCWuY96bCSc6AuhvkPpqI"/>
+  </RequestPayload>
+ </BatchItem>
+</RequestMessage>
+*/
+#define GET_REQUEST                                                           \
+   0x42, 0x00, 0x78, 0x01, 0x00, 0x00, 0x00, 0x88, 0x42, 0x00, 0x77, 0x01,    \
+      0x00, 0x00, 0x00, 0x38, 0x42, 0x00, 0x69, 0x01, 0x00, 0x00, 0x00, 0x20, \
+      0x42, 0x00, 0x6a, 0x02, 0x00, 0x00, 0x00, 0x04, 0x00, 0x00, 0x00, 0x01, \
+      0x00, 0x00, 0x00, 0x00, 0x42, 0x00, 0x6b, 0x02, 0x00, 0x00, 0x00, 0x04, \
+      0x00, 0x00, 0x00, 0x04, 0x00, 0x00, 0x00, 0x00, 0x42, 0x00, 0x0d, 0x02, \
+      0x00, 0x00, 0x00, 0x04, 0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0x00, 0x00, \
+      0x42, 0x00, 0x0f, 0x01, 0x00, 0x00, 0x00, 0x40, 0x42, 0x00, 0x5c, 0x05, \
+      0x00, 0x00, 0x00, 0x04, 0x00, 0x00, 0x00, 0x0a, 0x00, 0x00, 0x00, 0x00, \
+      0x42, 0x00, 0x79, 0x01, 0x00, 0x00, 0x00, 0x28, 0x42, 0x00, 0x94, 0x07, \
+      0x00, 0x00, 0x00, 0x20, 0x37, 0x46, 0x4a, 0x59, 0x76, 0x6e, 0x56, 0x36, \
+      0x58, 0x6b, 0x61, 0x55, 0x43, 0x57, 0x75, 0x59, 0x39, 0x36, 0x62, 0x43, \
+      0x53, 0x63, 0x36, 0x41, 0x75, 0x68, 0x76, 0x6b, 0x50, 0x70, 0x71, 0x49
+
+#define GET_UNIQUE_IDENTIFIER "7FJYvnV6XkaUCWuY96bCSc6AuhvkPpqI"
+
+void
+kms_kmip_request_get_test (void)
+{
+   kms_request_t *req;
+
+   uint8_t *actual_bytes;
+   uint32_t actual_len;
+   uint8_t expected_bytes[] = {GET_REQUEST};
+   uint32_t expected_len = sizeof (expected_bytes);
+
+   req = kms_kmip_request_get_new (NULL, GET_UNIQUE_IDENTIFIER);
```

Comment:
```suggestion
   static const char *const GET_UNIQUE_IDENTIFIER =
      "7FJYvnV6XkaUCWuY96bCSc6AuhvkPpqI";

   req = kms_kmip_request_get_new (NULL, GET_UNIQUE_IDENTIFIER);
```
I recommend using core language over preprocessor when able.

Reply:
Done, that seems sensible to me. I assume the reason is for additional type safety.